### DISCUSSION TITLE
Android Persisted Notification For GPS Geofencing

### DISCUSSION THREAD

**Discussion by SharpMobileCode** [shinyorg/shiny#854] [Q&A] [answered]
Hello Community,

I'm integrating geofencing an existing application and am seeing a persistent Notification on the Android notifications center.  The notification is titled "Shiny Service" and message says "Shiny service is continuing to process data in the background." (see image below)

![image](https://user-images.githubusercontent.com/8688589/146269001-d53c7761-2990-45fe-8617-7f98aadd2709.png)

I can't figure out a way to suppress this notification from my app.  It does seem off that it will have my Application Name, but make references to "Shiny".  If I present this to my business shareholders, they'll question what this notification is about and why can't they dismiss it.  I'm sure the end users will also question this as well.

If it helps my Startup looks like this:
```csharp
public override void ConfigureServices(IServiceCollection services, IPlatform platform)
{
    services.UseGpsDirectGeofencing<MyGeofenceDelegate>();
    services.UseNotifications();
}
```

Anyone have any idea how to either suppress it or make it go away?

**Comment by aritchie** [comment, marked as answer]
If you are going to use realtime GPS in the background, the persistent notification has to stay.  You can change it by resolving IGpsManager.  

var gps = ShinyHost.Resolve<IGpsManager>();
gps.Title = "Whatever";
gps.Message = "Whatever";

**Reply by papnak** [reply]
How can I customize Icon, Title and Message in Maui?

**Reply by aritchie** [reply]
You can do something like this on your gps delegate

```csharp
using Shiny.Locations;

namespace Project1.Delegates;

  public partial class MyGpsDelegate : IGpsDelegate
  {
      public MyGpsDelegate()
      {
      }


      public Task OnReading(GpsReading reading) => Task.CompletedTask;
  }

#if ANDROID
public partial class MyGpsDelegate : IAndroidForegroundServiceDelegate
{
    public void Configure(AndroidX.Core.App.NotificationCompat.Builder builder)
    {
        // TODO: customize your android notification here
    }
}
#endif
```
